src/ShadowMem.cc

```diff
@@ -275,14 +272,18 @@ class ShadowMemImpl : public InstVisitor<ShadowMemImpl> {
   /// \brief A map from DsaNode to its numeric id
   NodeIdMap m_nodeIds;
 
+  // Associate a shadow instruction to its Cell. Used by ShadowMem
+  // clients (e.g., VCGen).
+  llvm::DenseMap<const CallInst *, dsa::Cell> m_shadowMemInstToCell; 
```

Comment:
Why is this necessary? A `dsa` cell can be easily retrieved from the `CallInst` by moving to the previous instruction.
This is also more robust, in case dsa analysis is re-run.

Reply:
I agree but not sure you can always do that. How can you get the cell from a`shadow.mem.init` or `shadow.mem.arg.ref` instruction ?